### DISCUSSION TITLE
Target PMD 7.0 release date?

### DISCUSSION THREAD

**Discussion by gredler** [pmd/pmd#4789] [Q&A] [answered]
Apologies if I missed the information somewhere, but I have not been able to find a target release date for the final PMD 7.0 release. I'm planning to start using Java 21 soon, and it would help me to know when a non-RC PMD 7 release will be ready (specifically via Gradle). Is there a target release date (or month)? Thanks!

**Comment by jsotuyod** [comment]
I fear we haven't got a fixed schedule, as it depends greatly on our availability. There are slow months and some very productive ones.

The next release will probably be the final one. [We don't plan any more RCs](https://github.com/pmd/pmd/issues/3898), and what's left is mostly documentation (plenty of PRs already under review) and code cleanups (mostly removing deprecated stuff, or moving around classes that in PMD 6 we marked as internal).

The rules and general code base won't change, and the tool itself has proved to be stable enough, you should be able to reliably use it as it is.

Do remember however, that unless Gradle itself updates it's plugin you will have to manually add the `pmd-ant` dependency to the pmd configuration, regardless of PMD being an rc or not.

**Comment by gredler** [comment]
OK, thanks for the information. I'll keep an eye on the website over the next few months. Have a great day!

**Comment by adangel** [comment, marked as answer]
The target release date for PMD 7 will be March, 22nd 2024.
